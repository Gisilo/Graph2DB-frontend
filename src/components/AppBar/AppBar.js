import React from 'react';

import { Navbar, Nav, Button } from 'react-bootstrap/';

export class AppBar extends React.Component {

  render() {

    return (
      <Navbar bg="primary" expand="lg" className="shadow">
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav>
            <Nav.Link href="#home">Home</Nav.Link>
            <Nav.Link href="#project">New Project</Nav.Link>
            <Nav.Link href="#docs">Docs</Nav.Link>
          </Nav>
          <Navbar.Brand href="#home" className="mx-auto" style={{"fontFamily":"Pacifico-Regular"}}>
            Grask
          </Navbar.Brand>
          <Button variant="outline-secondary" style={{ "borderRadius": "35px" }} className="ml-2  ">Log In</Button>{' '}
          <Button variant="info" style={{ "borderRadius": "35px" }} className="ml-2">Sign Up</Button>{' '}
        </Navbar.Collapse>
      </Navbar>
    );

  }
}
