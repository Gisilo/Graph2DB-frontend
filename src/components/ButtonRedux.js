import React from 'react';
import { Button } from 'react-bootstrap';

import gql from 'graphql-tag'
import { withApollo } from 'react-apollo';
import { Mutation } from "react-apollo";

const query = gql`mutation CreateGrabitByName{
					createGrabit(
					input: {
					  nameProject: "prova_graph_88"
					  graph: "$newGraph"
					  
					}
					 ){grabit{
					nameProject
					graph
					  }
				  }
				  
					}`;



class SaveButton extends React.Component {

	saveWork = () => {
		let promise = this.props.client.query({
			query: query,
		});
		promise.then(
			(success) => console.log("suc", success),
			(error) => console.log("err", error))
	};



	render() {
		return (
			<Mutation mutation={query}>
				{(mutate, {data}) =>
					<Button onClick={() => {mutate({ variables: { newGraph: this.props.trigger() } });
					console.log("data", data)}} className="px-5" variant={this.props.variant}>
						{this.props.text}
					</Button>
				}
			</Mutation>

		);
	}

}

export default withApollo(SaveButton);
