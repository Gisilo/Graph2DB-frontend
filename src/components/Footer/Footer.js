import React from 'react';

export class Footer extends React.Component {


	render() {
		return (
			<footer
				// style={{
				// 	boxShadow: "0 -8px 16px rgba(0,0,0,.15)"
				// }}
				className="bg-primary container-fluid text-center text-md-left font-small info pt-2 shadow-lg">

				<div className="row justify-content-md-around mt-2">

					<div className="col-md-6">

						<h5 className="text-uppercase">Gisilo</h5>
						<p>Slogan to be found? Or something else</p>

					</div>

					<div className="col-md-3 mb-md-0 mb-1">

						<h5 className="text-uppercase">Connect</h5>

						<ul className="list-unstyled">
							<li href="#!">Meet the team</li>
							<li href="#!">Contacts Us</li>
							<li>TODO: riga con icone github e altri social</li>
						</ul>
					</div>

				</div>

				<div className="footer-copyright text-center pb-2">Â© 2020 Copyright: Gisilo</div>
			</footer>
		);
	}

}
