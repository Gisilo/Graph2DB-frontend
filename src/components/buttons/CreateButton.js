import React from 'react';
import gql from 'graphql-tag'
import { Button, Modal } from 'react-bootstrap';
import { InputTitleLeft, TextAreaTitleLeft, } from '../inputs';
import { Formik, ErrorMessage, Form } from 'formik';
import { useMutation } from 'react-apollo';

const CREATE_QUERY = gql`
	mutation CreateGrabit($nameGrabit: String!, $descr: String) {
		createGrabit(input: {nameProject: $nameGrabit, description: $descr}) {
			grabit {
		  		id
		  		nameProject
		  		description
		  	}
		}
	}`;

function MyVerticallyCenteredModal(props) {

	const [addGrabit] = useMutation(CREATE_QUERY);

	return (
		<Modal
			{...props}
			size="lg"
			aria-labelledby="contained-modal-title-vcenter"
			centered>

			<Modal.Header closeButton>
				<Modal.Title id="contained-modal-title-vcenter">
					Create new Grabit
		  		</Modal.Title>
			</Modal.Header>
			<Modal.Body>
				<Formik
					initialValues={{ grabitName: "", description: "" }}
					validate={values => {
						const errors = {};
						if (!values.grabitName) {
							errors.grabitName = 'Required';
						}
						return errors;
					}}
					onSubmit={(data, { setSubmitting }) => {
						setSubmitting(true);
						console.log("submit: ", data);
						// GRAPHQL REQUEST
						addGrabit({
							variables: {
								nameGrabit: data.grabitName,
								descr: data.description
							}
						}).then((success) => console.log('success', success), (error) => console.log('error', error));
						setSubmitting(false);
					}}>
					{({ values, isSubmitting }) => (
						<Form>
							<ErrorMessage name="grabitName" component="div" />
							<InputTitleLeft id="ig1" title="Grabit Name" placeholder="Grabit Name" name="grabitName" />
							<TextAreaTitleLeft id="ig4" title="Description" placeholder="Description (Optional)" name="description" />

							<Button disabled={isSubmitting} type="submit" variant="primary">Create</Button>
						</Form>
					)}
				</Formik>
			</Modal.Body>
		</Modal>
	);
}


function CreateButton() {
	const [modalShow, setModalShow] = React.useState(false);

	return (
		<>
			<Button variant="primary" onClick={() => setModalShow(true)}>
				New Project
			</Button>

			<MyVerticallyCenteredModal show={modalShow} onHide={() => setModalShow(false)}/>
		</>
	);
}


export default CreateButton;


