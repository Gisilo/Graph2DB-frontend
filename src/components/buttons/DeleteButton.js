import React from 'react';
import gql from 'graphql-tag'
import { Button } from 'react-bootstrap';
import { useMutation } from 'react-apollo';

import { DELETE_QUERY } from '../../costants/queries'


function DeleteButton(props){

    const [deleteGrabitMutation] = useMutation(DELETE_QUERY);

    const deleteGrabit = () => {

        deleteGrabitMutation({
            variables: {
                nameGrabit: "simone",
            }
        }).then((success) => console.log('success', success), (error) => console.log('error', error));
    };

    return (
        <Button onClick={deleteGrabit} className="px-5" variant={props.variant}>
            {props.text}
        </Button>

    );
}

export default DeleteButton;
