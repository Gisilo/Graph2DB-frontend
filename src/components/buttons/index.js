export {default as CreateButton} from './CreateButton.js';
export {default as DeleteButton} from './DeleteButton.js';
export {default as LoadButton} from './LoadButton.js';
export {default as SaveButton} from './SaveButton.js';
