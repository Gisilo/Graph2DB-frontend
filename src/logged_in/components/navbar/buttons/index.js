export {default as DashboardButton} from './DashboardButton';
export {default as EditorButton} from './EditorButton';
export {default as FeedbackButton} from './FeedbackButton';
export {default as LogoutButton} from './LogoutButton';
export {default as SchemaPageButton} from './SchemaPageButton';
export {default as SettingsButton} from './SettingsButton';