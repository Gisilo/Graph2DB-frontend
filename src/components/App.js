import React from 'react';

import { EditorScreen } from "./screens/EditorScreen";
import { LandingPage } from "./screens/LandingPage";
import NavBar from "./layout/navbar/NavBar";
import Footer from "./layout/footer/Footer";
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { Grid, TabPanel } from '@material-ui/core';
import Root from './layout/Root';


function App() {
    return (
        <Router>
            <Grid container>
                <Grid item xs={12}>
                    <Switch>
                        <Route path="/">
                            <Root/>
                        </Route>
                        <Route path="/editor">
                            <EditorScreen />
                        </Route>
                    </Switch>
                </Grid>
            </Grid>
        </Router>
    );
}

export default App;