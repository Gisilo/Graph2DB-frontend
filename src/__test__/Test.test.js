import React from 'react';
import { create } from "react-test-renderer";
import { MockedProvider } from '@apollo/react-testing';
import {AppBar} from "../components/AppBar/AppBar";
import {Footer} from "../components/Footer/Footer";
import { SaveButton, CreateButton, DeleteButton, LoadButton} from '../components/buttons';



describe('Button Tests', () => {

    test('SaveButton matches the snapshot', () => {
        const button = create(
            <MockedProvider addTypename={false}>
                <SaveButton />
            </MockedProvider>
        );
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('DeleteButton matches the snapshot', () => {
        const button = create(
            <MockedProvider addTypename={false}>
                <DeleteButton />
            </MockedProvider>
        );
        expect(button.toJSON()).toMatchSnapshot();
    });

    test('LoadButton matches the snapshot', () => {
        const button = create(
            <MockedProvider addTypename={false}>
                <LoadButton />
            </MockedProvider>
        );
        expect(button.toJSON()).toMatchSnapshot();
    });

    test("CreateButton matches the snapshot", () => {
        const button = create(
            <MockedProvider>
                <CreateButton />
            </MockedProvider>
        );
        expect(button.toJSON()).toMatchSnapshot();
    });
});


describe('AppBar Tests', () => {
    test("AppBar matches the snapshot", () => {
        const bar = create(
            <MockedProvider>
                <AppBar />
            </MockedProvider>
        );
        expect(bar.toJSON()).toMatchSnapshot();
    });
});


describe('Footer Tests', () => {
    test("Footer matches the snapshot", () => {
        const footer = create(
            <MockedProvider>
                <Footer />
            </MockedProvider>
        );
        expect(footer.toJSON()).toMatchSnapshot();
    });
});
