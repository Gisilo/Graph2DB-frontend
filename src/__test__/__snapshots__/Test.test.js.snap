// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Tests AppBar matches the snapshot 1`] = `
<nav
  className="shadow navbar navbar-expand-lg navbar-light bg-primary"
>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="#home"
        href="#home"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rb-event-key="#project"
        href="#project"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        New Project
      </a>
      <a
        className="nav-link"
        data-rb-event-key="#docs"
        href="#docs"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Docs
      </a>
    </div>
    <a
      className="mx-auto navbar-brand"
      href="#home"
      style={
        Object {
          "fontFamily": "Pacifico-Regular",
        }
      }
    >
      Grask
    </a>
    <button
      className="ml-2   btn btn-outline-secondary"
      disabled={false}
      style={
        Object {
          "borderRadius": "35px",
        }
      }
      type="button"
    >
      Log In
    </button>
     
    <button
      className="ml-2 btn btn-info"
      disabled={false}
      style={
        Object {
          "borderRadius": "35px",
        }
      }
      type="button"
    >
      Sign Up
    </button>
     
  </div>
</nav>
`;

exports[`Button Tests CreateButton matches the snapshot 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  New Project
</button>
`;

exports[`Button Tests DeleteButton matches the snapshot 1`] = `
<button
  className="px-5 btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button Tests LoadButton matches the snapshot 1`] = `
<button
  className="px-5 btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button Tests SaveButton matches the snapshot 1`] = `
<button
  className="px-5 btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Footer Tests Footer matches the snapshot 1`] = `
<footer
  className="bg-primary container-fluid text-center text-md-left font-small info pt-2 shadow-lg"
>
  <div
    className="row justify-content-md-around mt-2"
  >
    <div
      className="col-md-6"
    >
      <h5
        className="text-uppercase"
      >
        Gisilo
      </h5>
      <p>
        Slogan to be found? Or something else
      </p>
    </div>
    <div
      className="col-md-3 mb-md-0 mb-1"
    >
      <h5
        className="text-uppercase"
      >
        Connect
      </h5>
      <ul
        className="list-unstyled"
      >
        <li
          href="#!"
        >
          Meet the team
        </li>
        <li
          href="#!"
        >
          Contacts Us
        </li>
        <li>
          TODO: riga con icone github e altri social
        </li>
      </ul>
    </div>
  </div>
  <div
    className="footer-copyright text-center pb-2"
  >
    Â© 2020 Copyright: Gisilo
  </div>
</footer>
`;
